core/divesitehelpers.cpp

```diff
@@ -48,119 +44,116 @@ void reverseGeoLookup(degrees_t latitude, degrees_t longitude, taxonomy_data *ta
 	std::unique_ptr<QNetworkReply> reply(rgl.get(request));
 	timer.setSingleShot(true);
 	QObject::connect(&*reply, SIGNAL(finished()), &loop, SLOT(quit()));
-	timer.start(5000);   // 5 secs. timeout
+	timer.start(msTimeout);
 	loop.exec();
 
 	if (timer.isActive()) {
 		timer.stop();
 		if (reply->error() > 0) {
-			report_error("got error accessing geonames.org: %s", qPrintable(reply->errorString()));
-			return;
+			report_error("got error accessing %s: %s", qPrintable(url), qPrintable(reply->errorString()));
+			goto done;
```

Comment:
This makes code less readable, not more. Early return is idiomatic C++, please keep it.